byzcoin/service.go

```diff
@@ -1130,12 +1130,6 @@ func (s *Service) createNewBlock(scID skipchain.SkipBlockID, r *onet.Roster, tx
 		return nil, xerrors.Errorf("storing block: %v", err)
 	}
 
-	// State changes are cached only when the block is confirmed
-	err = s.stateChangeStorage.append(scs, ssbReply.Latest)
```

Comment:
Is there another call that append the states ?

Reply:
Yes, in `byzcoin.Service.updateTrieCallback`. This is called when the block from `StoreSkipblockInternal` is received by the node.